timApp/static/scripts/tim/folder/copy-folder.component.ts

```diff
@@ -135,14 +151,45 @@ export class CopyFolderComponent implements OnInit {
     }
 
     ngOnInit() {
-        this.copyFolderPath = this.item.path;
+        if (this.copyTo) {
+            this.copyFolderPath = this.copyTo;
+        } else {
+            this.copyFolderPath = this.currentItem.path;
+        }
+        if (this.copyFrom) {
+            this.fetchItemBasicInfo(this.copyFrom);
+        }
+    }
+
+    private async fetchItemBasicInfo(itemPath: string) {
+        const r = await toPromise(
+            this.http.get<{id: number; location: string; short_name: string}>(
+                `/itemInfo/${itemPath}`
+            )
+        );
+        if (!r.ok) {
+            return;
+        } else {
+            this.currentItem = {
+                id: r.result.id,
+                name: r.result.short_name,
+                path: r.result.location,
+            };
+            this.sourcePath =
+                this.currentItem?.path + "/" + this.currentItem?.name;
+        }
     }
 
-    async copyFolderPreview(path: string, exclude: string) {
+    async copyFolderPreview(
+        path: string,
+        exclude: string,
+        itemId: number = this.currentItem.id
+    ) {
```

Comment:
Tarvitseeko enää tätä lisäparametria `itemId`, eli voisiko senkin palauttaa käyttämään `this.currentItem`ia kuten ennenkin?